packages/facebook-auth/README.md

```diff
@@ -0,0 +1,162 @@
+	}
+)
+
+export default app
+
+```
+
+### facebookAuth
+```
+import { Hono } from  'hono'
+import { FacebookAuthVariables, facebookAuth } from  '@hono/facebook-auth'
+
+const  app  =  new  Hono<{ Variables:  FacebookAuthVariables }>()
+
+app.use(
+  '/facebook',
+  facebookAuth({
+    client_id:  Bun.env.FACEBOOK_ID,
+    client_secret:  Bun.env.FACEBOOK_SECRET,
+    response_type: ['token'],
+		scope: ['email', 'public_profile'],
+		fields: ['email', 'id', 'first_name', 'last_name', 'middle_name', 'name', 'picture', 'short_name'],
+		include_granted_scopes:  true
+  })
+)
+
+export  default  app
+```
+#### Parameters
+- `client_id`:
+ 	- Type: `string`.
+	- `Required`.
+	-  Your app client ID. You can find this value in the App Dashboard [Dashboard page](https://developers.facebook.com/apps). <br />When developing **Cloudflare Workers**, there's no need to send this parameter. Just declare it in the `wrangler.toml` file as `FACEBOOK_ID=`.
+- `client_secret`:
+	- Type: `string`.
+	- `Required`.
+	- Your app client secret. You can find this value in the App Dashboard [Dashboard page](https://developers.facebook.com/apps). <br />When developing **Cloudflare Workers**, there's no need to send this parameter. Just declare it in the `wrangler.toml` file as `FACEBOOK_SECRET=`.
+	> Do not share your client secret to ensure the security of your app.
+- `scope`:
+	- Type: `string[]`.
+	- `Required`.
+	- Set of **permissions** to request the user's authorization to access your app for retrieving user information and performing actions on their behalf.<br /> Review all the scopes Facebook offers for utilizing their API on the [Permissions page](https://developers.facebook.com/docs/permissions/). 
+	> If your app is not **verified** by Facebook, the accessible scopes for your app are significantly **limited**.
+- `response_type`:
+	- Type: `string[]`.
+	- `Required`.
+	- Choose the **OAuth flow** for the authentication process. The available options are:
+		- `code` for a more secure flow, applicable when the app's authentication flow is managed by the **server side**.
+		- `token` for quick access to the token, suitable when the app's authentication flow will be managed by the **client side**.
+		- `code%20token`:  A combination of both the code and access token received as parameters on the url.
+- `include_granted_scopes`:
+	- Type: `boolean`.
+	- `Optional`.
+	- To obtain the scopes for which the user has **granted permission**, set this value to `true`. This will allow you to implement a flow in cases where your app cannot access crucial permissions and needs to request them again, among other use cases.  Value `false` by default.
+- `fields`:
+	- Type: `string[]`.
+	- Fields you request from the Facebook API to be sent once the user has logged in. You can find a comprehensive reference for all the fields you can request on the [Facebook User Reference page](https://developers.facebook.com/docs/graph-api/reference/user/#fields).
+- `state`:
```

Comment:
`state` isn't a secret. It just needs to be different time. You defeat the purpose of using `state` if you use the same value every time. A new `state` should be generated and stored (usually as a cookie) every time the user is redirected to the provider's login page.

https://pilcrow.vercel.app/blog/oauth-guide#authorization-url

Reply:
Didn't thought about it, I will handle the creation and storage of the `state` internally. Thanks!